### PR TITLE
Content munge devportal

### PR DESCRIPTION
**Pull request by laurapacilio**
Updates the content in preparation for the developer portal migration. Specifically:
- Removes mentions of HashiCorp Learn, as this will no longer exist
- Corrects any mentions of packer.io or the platform in a way that will feel awkward
- Removes mentions of the documentation sidebar or navigation (e.g. click the link in the left sidebar navigation)
- Removes mentions of learn "collections" where possible
- Checks for URLs to packer.io or learn that are user-facing (found none)
- Checks for absolute links to docs (found none)

### CODE REVIEW

File: website/content/docs/templates/hcl_templates/blocks/index.mdx [hunk 1]
```diff
@@ -27,7 +27,7 @@ The most important blocks can be broken into a couple of major types:
   HCL functions or data sources, or composited from variables created in the
   variables blocks.
 
-Use the sidebar to navigate to detailed documentation for each of these blocks.
+The documentation contains a page for each block type.
```
Comment (laurapacilio) [inline website/content/docs/templates/hcl_templates/blocks/index.mdx:30]:
```suggestion
The documentation contains information for each block type.
```

File: website/content/docs/provisioners/index.mdx [hunk 2]
```diff
@@ -7,15 +7,13 @@ page_title: Provisioners
...
 - installing packages
 - patching the kernel
 - creating users
 - downloading application code
 
-See the [`provisioner`](/docs/templates/hcl_templates/blocks/build/provisioner) block documentation to learn more
-about working with provisioners. For information on an individual provisioner,
-choose it from the sidebar.
+Refer to the [`provisioner`](/docs/templates/hcl_templates/blocks/build/provisioner) block documentation to learn more
+about working with provisioners. The documentation includes a page for each type of provisioner.
```
Comment (laurapacilio) [inline website/content/docs/provisioners/index.mdx:19]:
```suggestion
about working with provisioners. The documentation includes details about each type of provisioner.
```

File: website/content/docs/datasources/index.mdx [hunk 3]
```diff
@@ -7,12 +7,9 @@ page_title: Data Sources
...
 # Data Sources
 
-Data sources allow data to be fetched for use in Packer configuration. Use of data sources
-allows a build to use information defined outside of Packer.
+Data sources let Packer fetch data to use in a template, including information defined outside of Packer.
 
-See the [`data`](/docs/templates/hcl_templates/datasources) block documentation to learn more
-about working with data sources. For information on an individual data source,
-choose it from the sidebar.
+Refer to the [`data`](/docs/templates/hcl_templates/datasources) block documentation to learn more about working with data sources. The documentation also contains a page for each type of data source.
```
Comment (laurapacilio) [inline website/content/docs/datasources/index.mdx:12]:
```suggestion
Refer to the [`data`](/docs/templates/hcl_templates/datasources) block documentation to learn more about working with data sources. The documentation also contains details about each type of data source.
```

**Review by nywilken — APPROVED** [review]
These changes look good to me.

**Pull request merged**